static/app/views/dashboards/widgets/timeSeriesWidget/plottables/continuousTimeSeries.tsx

```diff
@@ -79,6 +79,22 @@ export abstract class ContinuousTimeSeries<
     return this.#timestamps.at(-1) ?? null;
   }
 
+  /**
+   * Shallow clones `timeSeries` and constrains `timeSeries` data to be between
+   * boundary datetime (if provided).
+   */
+  constrainTimeSeries(boundaryStart: Date | null, boundaryEnd: Date | null) {
```

Comment:
Allowing null in case you don't care about a certain boundary, but no strong opinion here if we want to disallow it.